app/screens/channel_info/index.js

```diff
@@ -100,8 +106,17 @@ function mapStateToProps(state) {
         timeZone = getUserCurrentTimezone(currentUser.timezone);
     }
 
+    let canUnarchiveChannel = false;
+    if (hasNewPermissions(state) && isMinimumServerVersion(serverVersion, 5, 20)) {
+        canUnarchiveChannel = haveITeamPermission(state, {
```

Comment:
@wiersgallak I do not know much about permissions, but are we only able to archive a channel if we have team/system admin permissions? are there any other permissions/role to check? what about if I'm the creator/admin of the channel ?

Reply:
@enahum 

> but are we only able to archive a channel if we have team/system admin permissions?

By default, any user can archive a channel. In E10 and E20, you can restrict it to channel, team and/or system admins only via the permission schemes.

For unarchived channels, looks like the API call requires `manage_team` permission for the team of the channel: https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1%7Bchannel_id%7D~1restore%2Fpost

I don't understand why, but that's beyond the scope of this PR. In my opinion it should be any member of the channel unless restricted via the E10/E20 permission schemes, like for archiving a channel.